commands/serve_command.go

```diff
@@ -79,18 +82,43 @@ func (c *ServeCommand) Run(args []string) int {
 	srv := langserver.NewLangServer(ctx, handlers.NewSession)
 	srv.SetLogger(logger)
 
+	if c.tfExecPath != "" {
+		path := c.tfExecPath
+
+		logger.Printf("Setting Terraform exec path to %q", path)
+
+		// just some sanity checking here, no need to get too specific otherwise will be complex cross-OS
+		if !filepath.IsAbs(path) {
+			c.Ui.Error(fmt.Sprintf("Expected absolute path for Terraform binary, got %q", path))
+			return 1
+		}
+		stat, err := os.Stat(path)
+		if err != nil {
+			c.Ui.Error(fmt.Sprintf("Unable to find Terraform binary: %s", err))
+			return 1
+		}
+		if stat.IsDir() {
+			c.Ui.Error(fmt.Sprintf("Expected a Terraform binary, got a directory: %q", path))
+			return 1
+		}
```

Comment:
Total nitpick, but I assume the above lines could make a `validateTfExecPath(string) error` function.